backend/src/apiserver/storage/run_store.go

```diff
@@ -350,6 +353,25 @@ func (s *RunStore) CreateRun(r *model.RunDetail) (*model.RunDetail, error) {
 }
 
 func (s *RunStore) UpdateRun(runID string, condition string, workflowRuntimeManifest string) (err error) {
+	tx, err := s.db.DB.Begin()
+	if err != nil {
+		return util.NewInternalServerError(err, "transaction creation failed")
+	}
+
+	row := tx.QueryRow("SELECT WorkflowRuntimeManifest FROM run_details WHERE UUID = ? FOR UPDATE", runID)
+	var storedManifest string
+	if err := row.Scan(&storedManifest); err != nil {
+		tx.Rollback()
+		return util.NewResourceNotFoundError("Run", runID)
```

Comment:
We lose the error message here. 

Additionally, isn't it possible that the error is not a RESOURCE_NOT_FOUND?